CONTRIBUTING.rst

```diff
@@ -0,0 +1,162 @@
+Contributing to the Jupyter Notebook
+====================================
+
+TODO: a welcoming sentence
+
+General Guidelines
+------------------
+
+For general documentation about contributing to Jupyter projects, see the
+`Project Jupyter Contributor Documentation`__.
+
+__ http://jupyter.readthedocs.org/#contributor-documentation
+
+
+Setting Up a Development Environment
+------------------------------------
+
+If you have already installed the dependencies mentioned below, the following
+steps should get you going::
+
+    pip install setuptools pip --upgrade --user
+    git clone https://github.com/jupyter/notebook
+    cd notebook
+    pip install -e . --user
+
+If you want this to be available for all users of your system (assuming you
```

Comment:
OK.

I just tried it in a Python 3 virtualenv and it seems that the `--user` option also works there, so I'll remove the comment about the virtualenv here.